### PR TITLE
OSD device replacement fixes for ocs 4.8

### PR DESCRIPTION
**Pull request by shreekarSS**
- Fixes https://github.com/red-hat-storage/ocs-ci/issues/4276

 - Code for osd replacement is moved to osd_operations file

### CODE REVIEW

**Comment by yitzhak12** [comment]
I would divide the code into smaller functions. For example, instead of 

> osd_pod = [        osd_pod        for osd_pod in osd_pods        if osd_pod.get()        .get("metadata")        .get("labels")        .get(constants.CEPH_ROOK_IO_PVC_LABEL)        == claim_name    ][0]

You create a new function `get_osd_pod`, that returns the osd pod. 
or instead

> osd_prepare_pod = [            pod            for pod in osd_prepare_pods            if pod.get()            .get("metadata")            .get("labels")            .get(constants.CEPH_ROOK_IO_PVC_LABEL)            == claim_name        ][0]

You can create a new function `get_osd_prepare_pod`, that returns the osd prepare pod. 
And the same thing with all the small code snippets that we have here in the file.
That way, it will be much easier to fix or to improve the existing code. Let me know if it sounds good to you.

File: ocs_ci/ocs/osd_operations.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    get_operator_pods,
+    get_osd_prepare_pods,
+    get_osd_pod_id,
+    run_osd_removal_job,
+    verify_osd_removal_job_completed_successfully,
+    delete_osd_removal_job,
+)
+from ocs_ci.helpers.sanity_helpers import Sanity
+
+import logging
```
Comment (AviadP) [inline ocs_ci/ocs/osd_operations.py:23]: minor comment: plz sort imports according as:
import builtin lib
import 3rd party lib
import local lib

Comment (shreekarSS) [reply ocs_ci/ocs/osd_operations.py:23]: Ack

File: ocs_ci/ocs/osd_operations.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+
+logger = logging.getLogger(__name__)
+
+
+def osd_device_replacement(nodes):
+    """
+    Replacing randomly picked osd device
+    Args:
+        node (OCS): The OCS object representing the node
```
Comment (yitzhak12) [inline ocs_ci/ocs/osd_operations.py:33]: Just one small comment... Need to change the name to `nodes`.

Comment (shreekarSS) [reply ocs_ci/ocs/osd_operations.py:33]: ack

**Comment by shreekarSS** [comment]
> I would divide the code into smaller functions. For example, instead of
> 
> > osd_pod = [        osd_pod        for osd_pod in osd_pods        if osd_pod.get()        .get("metadata")        .get("labels")        .get(constants.CEPH_ROOK_IO_PVC_LABEL)        == claim_name    ][0]
> 
> You create a new function `get_osd_pod`, that returns the osd pod.
> or instead
> 
> > osd_prepare_pod = [            pod            for pod in osd_prepare_pods            if pod.get()            .get("metadata")            .get("labels")            .get(constants.CEPH_ROOK_IO_PVC_LABEL)            == claim_name        ][0]
> 
> You can create a new function `get_osd_prepare_pod`, that returns the osd prepare pod.
> And the same thing with all the small code snippets that we have here in the file.
> That way, it will be much easier to fix or to improve the existing code. Let me know if it sounds good to you.

I think its not really needed, just 1 for loop

**Comment by yitzhak12** [comment]
@shreekarSS It's not mandatory. Whatever you feel comfortable with.

**Review by yitzhak12 — APPROVED** [review]

**Review by PrasadDesala — APPROVED** [review]

**Pull request merged**